app/middleware/getServices.js

```diff
@@ -0,0 +1,70 @@
+const VError = require('verror').VError;
+const log = require('../lib/logger');
+const elasticsearchClient = require('../lib/elasticsearchClient');
+const esGeoQueryBuilder = require('../lib/esGeoQueryBuilder');
+const esGetServiceHistogram = require('../lib/promHistograms').esGetServices;
+const esQueryLabelName = require('../lib/constants').PROM_ES_QUERY_LABEL_NAME;
+const constants = require('../lib/constants');
+
+function handleError(error, next) {
+  const errMsg = 'Error with ES';
+
+  log.error({ err: new VError(error.stack, errMsg) }, errMsg);
+  next(error);
+}
+
+function mapResults(results, res) {
+  res.locals.services = results.hits.hits.map((result) => {
+    // eslint-disable-next-line no-underscore-dangle
+    const service = result._source;
+
+    if (service) {
+      if (result.sort) {
+        service.distance = result.sort[0];
+      }
+    }
+
+    return service;
+  });
+}
+
+function getEsQuery(postcodeLocationDetails, searchType, size) {
+  const esQuery = {
+    label: searchType,
+    query: esGeoQueryBuilder.build(postcodeLocationDetails.location, searchType, size),
+  };
+  return esQuery;
+}
+
+function getServices(req, res, next) {
+  const location = res.locals.location;
+  const resultsLimit = res.locals.RESULTS_LIMIT;
+  const postcodeLocationDetails = res.locals.postcodeLocationDetails;
+  const esQuery = getEsQuery(
+    postcodeLocationDetails,
+    constants.SEARCH_TYPES.sexperts,
+    resultsLimit
+  );
+  log.info('query ', esQuery.query);
```

Comment:
The signature for log methods is `method(obj, string)`. Can the message be updated to be a bit more informative too please?

Reply:
This was for me for debugging, will remove.